source/kernel/src/registry/mod.rs

```diff
@@ -89,6 +107,9 @@ pub struct UserResponse<U, E> {
 #[non_exhaustive]
 pub struct Envelope<P> {
     pub body: P,
+    service_id: u32,
+    client_id: u32,
+    request_id: u32,
```

Comment:
seems like it might eventually be worth introducing newtypes for `ServiceId`, `ClientId`, and `RequestId`, to ensure these aren't confusable? but we can do that in a follow-up.